src/main/java/com/denizenscript/depenizen/bukkit/objects/towny/TownTag.java

```diff
@@ -224,6 +226,30 @@ public static void registerTags() {
             return new ElementTag(object.town.getBoard());
         });
 
+        // <--[tag]
+        // @attribute <TownTag.members_by_rank[(<rank>)]>
+        // @returns ListTag
+        // @plugin Depenizen, Towny
+        // @description
+        // Returns a list of the town's members with a given rank. Players will be valid PlayerTag instances, non-players will be plaintext of the name.
+        // -->
+        tagProcessor.registerTag(ListTag.class, "members_by_rank", (attribute, object) -> {
+            String rank = attribute.getParam();
+            ListTag list = new ListTag();
+            for (Resident resident : object.town.getRank(rank)) {
+                if (resident.getUUID() != null) {
```

Comment:
Why does this tag have all this validation, but NationTag one doesn't?
Are the two Resident lists explicitly different, or is one of the two versions correct and the other wrong?

Reply:
This was because it's based off Assistants -I used the relevant portion of existing code in both; assuming there was a reason it was done in Depenizen originally; not a choice by myself - but I can change this to match the TownTag one if wanted; I'm not sure what the reasoning behind the difference in assistants code is; hence why I used it without 'correcting' it to potentially be wrong.